File: wdae/api/common/effect_types.py
'''
Created on Jul 28, 2015

@author: lubo
'''
import operator

EFFECT_TYPES = {
    "Nonsense": ["nonsense"],
    "Frame-shift": ["frame-shift"],
    "Splice-site": ["splice-site"],
    "Missense": ["missense"],
    "Non-frame-shift": ["no-frame-shift"],
    "noStart": ["noStart"],
    "noEnd": ["noEnd"],
    "Synonymous": ["synonymous"],
    "Non coding": ["non-coding"],
    "Intron": ["intron"],
    "Intergenic": ["intergenic"],
    "3'-UTR": ["3'UTR", "3'UTR-intron"],
    "5'-UTR": ["5'UTR", "5'UTR-intron"],
    "CNV": ["CNV+", "CNV-"],

}


EFFECT_GROUPS = {
    "coding": [
        "Nonsense",
        "Frame-shift",
        "Splice-site",
        "Missense",
        "Non-frame-shift",
        "noStart",
        "noEnd",
        "Synonymous",
    ],
    "noncoding": [
        "Non coding",
        "Intron",
        "Intergenic",
        "3'-UTR",
        "5'-UTR",
    ],
    "cnv": [
        "CNV+",
        "CNV-"
    ],
    "lgds": [
        "Nonsense",
        "Frame-shift",
        "Splice-site",
    ],
    "nonsynonymous": [
        "Nonsense",
        "Frame-shift",
        "Splice-site",
        "Missense",
        "Non-frame-shift",
        "noStart",
        "noEnd",
    ],
    "utrs": [
        "3'-UTR",
        "5'-UTR",
    ]

}


def build_effect_type_filter(data):
    if "effectTypes" not in data:
        return
    effects_string = data['effectTypes']
    effects = effects_string.split(',')
    result_effects = \
        reduce(operator.add,
               [EFFECT_TYPES[et] if et in EFFECT_TYPES else [et]
                for et in effects])
    data["effectTypes"] = ','.join(result_effects)

File: web_ui/src/app/effect-types/effect-types.ts
export const ALL: Set<string> = new Set([
  '3\'UTR',
  '3\'UTR-intron',
  '5\'UTR',
  '5\'UTR-intron',
  'frame-shift',
  'intergenic',
  'intron',
  'missense',
  'no-frame-shift',
  'no-frame-shift-newStop',
  'noEnd',
  'noStart',
  'non-coding',
  'non-coding-intron',
  'nonsense',
  'splice-site',
  'synonymous',
  'CDS',
  'CNV+',
  'CNV-',
]);

export const CODING: Set<string> = new Set([
  'nonsense',
  'frame-shift',
  'splice-site',
  'no-frame-shift-newStop',
  'missense',
  'no-frame-shift',
  'noStart',
  'noEnd',
  'synonymous',
]);

export const NONCODING: Set<string> = new Set([
  'non-coding',
  'intron',
  'intergenic',
  '3\'UTR',
  '5\'UTR',
]);

export const CNV: Set<string> = new Set([
  'CNV+',
  'CNV-'
]);

export const LGDS: Set<string> = new Set([
  'frame-shift',
  'nonsense',
  'splice-site',
  'no-frame-shift-newStop',
]);

export const NONSYNONYMOUS: Set<string> = new Set([
  'nonsense',
  'frame-shift',
  'splice-site',
  'no-frame-shift-newStop',
  'missense',
  'no-frame-shift',
  'noStart',
  'noEnd',
]);

export const UTRS: Set<string> = new Set([
  '3\'UTR',
  '5\'UTR',
]);

export const OTHER: Set<string> = new Set([
  '3\'UTR',
  '3\'UTR-intron',
  '5\'UTR',
  '5\'UTR-intron',
  'intergenic',
  'intron',
  'no-frame-shift',
  'noEnd',
  'noStart',
  'non-coding',
  'non-coding-intron',
  'CDS',
]);

export const GENOTYPE_BROWSER_INITIAL_VALUES: Set<string> = new Set([
  'nonsense',
  'frame-shift',
  'splice-site',
  'no-frame-shift-newStop',
]);
